### PR TITLE
Upgrade to Node 16

### LINKED ISSUE #410: Upgrade to node 16
Actions using **Node 12** are considered deprecated and encouraged to upgrade to **Node 16**.
For reference: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/

I wonder if all that is needed is just changing [this line](https://github.com/coursier/cache-action/blob/main/action.yml#L122).

```diff
runs:
-  using: 'node12'
+  using: 'node16'
  main: 'dist/restore/index.js'
  post: 'dist/save/index.js'
  post-if: 'success()'
```

And refreshing the `package-lock.json` using **Node 16**.

If so, I am more than happy to help.
But, if something else is required I would need assistance / guidance.

See also: https://github.com/coursier/setup-action/issues/281

### PR DESCRIPTION
**Pull request by BalmungSan**
Fixes #410

### CODE REVIEW

**Comment by daddykotex** [comment]
Oh wow, 8 hours ago. I had not realized that.

Good job, I was going to ping someone for review but we'll give them time to breathe!

**Review by daddykotex — APPROVED** [review]
Approving, but I have not reviewed the package-lock.json carefully.

File: src/restore.ts [hunk 1]
```diff
@@ -2,8 +2,8 @@ import * as cache from '@actions/cache'
 import * as core from '@actions/core'
 import * as exec from '@actions/exec'
 import {readFile, stat, unlink, writeFile} from 'fs'
-const glob = require('glob-all')
-const hashFiles = require('hash-files')
+import {glob} from 'glob-all'
+import {hashFiles} from 'hash-files'
```
Comment (alexarchambault) [inline src/restore.ts:6]: Neat!

**Comment by daddykotex** [comment]
can you approve the workflow @alexarchambault just so we have a green pipeline and see if this needs work?

**Pull request merged**